common/src/test/java/io/druid/collections/ReferenceCountingResourceHolderTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.collections;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ReferenceCountingResourceHolderTest
+{
+  @Test
+  public void testIdiomaticUsage()
+  {
+    // Smoke testing
+    for (int i = 0; i < 100; i++) {
+      runIdiomaticUsage();
+    }
+  }
+
+  private void runIdiomaticUsage()
+  {
+    final AtomicBoolean released = new AtomicBoolean(false);
+    final ReferenceCountingResourceHolder<Closeable> resourceHolder = makeReleasingHandler(released);
+    List<Thread> threads = new ArrayList<>();
+    for (int i = 0; i < 100; i++) {
+      Thread thread = new Thread() {
+        @Override
+        public void run()
+        {
+          try (Releaser r = resourceHolder.increment()) {
```

Comment:
Should there be a start gate here? or else on an overburdened CI system the threads could easily all start in sequence without actually racing anything